src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -1,14 +1,168 @@
 package cromwell.binding
 
+import cromwell.binding.AstTools.AstNodeName
+import cromwell.parser.WdlParser.{AstList, Ast, AstNode}
+import scala.collection.JavaConverters._
+
+
+object Scope {
+
+  /**
+   * Utility class to generate a ScopeTree from an Ast.
+   */
+  private class ScopeTreeGenerator {
+
+    /* Keep track of the # of created Scatter nodes. */
+    var scatterCount: Int = 0
+
+  /*
+  * Retrieve the list of AstNodes from the AST body.
+  * Return empty seq if body is null or is not a list.
+  */
+    private def getChildrenList(ast: Ast): Seq[AstNode] = {
+      val body: AstNode = ast.getAttribute("body")
+      if(body == null) Seq.empty
+      else {
+        body match {
+          case list: AstList => list.asScala.toVector
+          case _ => Seq.empty
+        }
+      }
+    }
+
+    /**
+     * Generate recursively all descendant Scopes for the workflow.
+     * Should only be called during apply with a root Ast
+     * @param node current working node
+     * @param namespaces namespaces
+     * @param tasks tasks
+     * @param wdlSyntaxErrorFormatter formatter
+     * @return Some(instance of Scope) if the node has a Scope equivalent, None otherwise
+     */
```

Comment:
the `@param` entries above are kinda boilerplate... can you either fill them out or remove them?